aerotech_101smc2/aerotech_101smc2/_aerotech.py

```diff
@@ -36,21 +38,22 @@ def get_state(self):
         state["value"] = self.value
         return state
 
-
-
-
     def _set_position(self, position):
         ...
 
+    def close(self):
+        self._serial_port.close()
+
+    def direct_serial_write(self, message):
+        self._busy = True
+        self._serial_port.write(message.encode())
 
     async def update_state(self):
-        """Continually monitor and update the current daemon state."""
-        # If there is no state to monitor continuously, delete this function
         while True:
-            # Perform any updates to internal state
-            self._busy = False
-            # There must be at least one `await` in this loop
-            # This one waits for something to trigger the "busy" state
-            # (Setting `self._busy = True)
-            # Otherwise, you can simply `await asyncio.sleep(0.01)`
-            await self._busy_sig.wait()
+            self._serial_port.write(b"get_status")
+            line = await self._serial_port.areadline()
+            self._busy = line != b"ready"
+        if self._busy:
```

Comment:
These lines should be indented (this was a problem on the web page resulting from accidental mixing of tabs and spaces...)

https://gitlab.com/yaq/python-yaq-fyi/-/merge_requests/4